src/fiatconnect/ReviewScreen.tsx

```diff
@@ -240,55 +241,168 @@ function ReceiveAmount({
   normalizedQuote: FiatConnectQuote
 }) {
   const { t } = useTranslation()
+  const exchangeRates = useSelector(localCurrencyExchangeRatesSelector)!
+
+  const { receiveDisplay } = getDisplayAmounts({
+    flow,
+    normalizedQuote,
+    exchangeRates,
+  })
   return (
     <View style={styles.receiveAmountContainer}>
       <LineItemRow
         style={styles.sectionMainTextContainer}
         textStyle={styles.sectionMainText}
         title={t('fiatConnectReviewScreen.receiveAmount')}
-        amount={
-          flow === CICOFlow.CashIn ? (
-            <TokenDisplay
-              amount={normalizedQuote.getCryptoAmount()}
-              currency={normalizedQuote.getCryptoType()}
-              showLocalAmount={false}
-              testID="amount-crypto"
-            />
-          ) : (
-            <CurrencyDisplay
-              amount={{
-                // The value here doesn't matter since the component will use `localAmount`
-                value: 0,
-                currencyCode: resolveCICOCurrency(normalizedQuote.getCryptoType()),
-                localAmount: {
-                  value: normalizedQuote.getFiatAmount(),
-                  currencyCode: normalizedQuote.getFiatType(),
-                  exchangeRate: 1,
-                },
-              }}
-              testID="amount-fiat"
-            />
-          )
-        }
+        amount={receiveDisplay('receive-amount')}
       />
     </View>
   )
 }
 
-function TransactionDetails({
+function CryptoAmount({
+  amount,
+  currency,
+  testID,
+}: {
+  amount: BigNumber.Value
+  currency: Currency
+  testID: string
+}) {
+  return (
+    <TokenDisplay amount={amount} currency={currency} showLocalAmount={false} testID={testID} />
+  )
+}
+
+function FiatAmount({
+  amount,
+  currency,
+  testID,
+  formatType,
+}: {
+  amount: BigNumber.Value
+  currency: string
+  testID: string
+  formatType?: FormatType
+}) {
+  return (
+    <CurrencyDisplay
+      amount={{
+        // The value and currencyCode here doesn't matter since the component will use `localAmount`
+        value: 0,
+        currencyCode: '',
+        localAmount: {
+          value: amount,
+          currencyCode: currency,
+          exchangeRate: 1,
+        },
+      }}
+      formatType={formatType}
+      testID={testID}
+    />
+  )
+}
+
+function getDisplayAmounts({
   flow,
   normalizedQuote,
+  exchangeRates,
 }: {
   flow: CICOFlow
   normalizedQuote: FiatConnectQuote
+  exchangeRates: {
+    cGLD: string | null
+    cUSD: string | null
+    cEUR: string | null
+  }
 }) {
-  const exchangeRates = useSelector(localCurrencyExchangeRatesSelector)!
+  const cryptoType = normalizedQuote.getCryptoType()
+  const fiatType = normalizedQuote.getFiatType()
+  if (flow === CICOFlow.CashOut) {
```

Comment:
I think some of the logic here could be condensed, and common stuff pulled out of the conditional? Looks to be a decent amount of duplication currently.